File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.cpp
class Solution {
public:

    long long maximumTripletValue(vector<int>& nums) {
        int n = nums.size();
        long long val = 0;
        for(int i = 0; i<n; i++){
            for(int j = i+1; j<n; j++){
                if(nums[i] - nums[j] <= 0)
                    continue;
                for(int k = j+1; k<n; k++){
                    long long tempVal =(long long)(nums[i]-nums[j])*nums[k];
                    val = max(val, tempVal);
                }
            }
        }
        return val;
    }
};

File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.java
class Solution {
    public long maximumTripletValue(int[] nums) {
        int n = nums.length;
        long ans = 0;
        
        int maxDiff = 0, maxI = 0;
        // calculate the max triplets
        for(int i = 0; i<n ; i++){
            ans = Math.max(ans, (long)maxDiff*nums[i]);
            maxDiff = Math.max(maxDiff, maxI - nums[i]);
            maxI = Math.max(maxI, nums[i]);
        }
        return ans;
    }
}